test/fixtures/s3-fixture/src/main/java/fixture/s3/junit/S3HttpFixtureRule.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package fixture.s3.junit;
+
+import fixture.s3.S3HttpFixture;
+import fixture.s3.S3HttpFixtureWithEC2;
+import fixture.s3.S3HttpFixtureWithECS;
+import fixture.s3.S3HttpFixtureWithSTS;
+import fixture.s3.S3HttpFixtureWithSessionToken;
+
+import org.junit.rules.ExternalResource;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class S3HttpFixtureRule extends ExternalResource {
+    private final boolean enabled;
+    private final List<Type> enabledFixtures;
+
+    private Map<Type, S3HttpFixture> configuredFixtures = new HashMap<>();
+
+    private void configureFixture(Type fixtureType) throws Exception {
+        InetSocketAddress inetSocketAddress = resolveLoopBackAddress();
+        int port = inetSocketAddress.getPort();
+
+        if (fixtureType == Type.S3Fixture) {
```

Comment:
This pattern is confusing to me. I think we should separate these out as individual fixtures, rather than have a single "fixture" that actually provides multiple functions. As my comment above asks, it's not clear these could ever be used concurrently in a single test anyway as they'd fail trying to bind to the same address.

Reply:
they do bind a different port. anyhow. I'm reworking this as suggested above